gamemodes/terrortown/gamemode/server/sv_corpse.lua

```diff
@@ -81,6 +95,20 @@ function GM:TTTCanIdentifyCorpse(ply, corpse)
 	return true
 end
 
+local function CanConfirmPlayer(deadply, ply, rag)
+	if hook.Run("TTT2ConfirmPlayer", deadply, ply, rag) == false then
+		return false
+	end
+
+	if cvDeteOnlyConfirm:GetBool() and ply:GetBaseRole() ~= ROLE_DETECTIVE then
+		LANG.Msg(ply, "confirm_detective_only", nil, MSG_MSTACK_WARN)
+
+		return false
+	end
```

Comment:
I feel like these two checks should be switched. If `Detective only confirm` is enabled that message should always show up, otherwise it would feel quite inconsistent.